meson: remove unnecessary stuff

emilazy opened the pull request:
Meson changes should be good if they build, but I’d like Randy’s feedback on the `stdenv` stuff.
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

emilazy committed:
meson: remove OpenLDAP dependency

No longer necessary now that the SDK retains the Apple OpenLDAP
fork headers.

emilazy committed:
meson: remove bitcode patch

Shouldn’t be necessary. We do still disable the test, because `ld64`
refuses to compile bitcode (and `ld-prime` just ignores the option);
Apple dropped support for bitcode quite a while ago.

emilazy committed:
meson: remove conditional `libxcrypt` dependency

We no longer ship Python < 3.9.

emilazy committed:
meson: remove old Darwin SDK pattern detritus

emilazy committed:
stdenv/darwin: remove OpenLDAP

This wasn’t being built in the bootstrap anyway. We still can’t
run the Meson tests because of issues with pkgconf, which I’ll try
to address another time.

toonn reviewed (approved):
LGTM, especially the net line count : )

emilazy committed:
apple-sdk: don’t remove LDAP framework headers

Apple ships a 13‐year‐old fork of OpenLDAP. The very few things
that use it, like Meson tests, have been adding dependencies on
our up‐to‐date OpenLDAP package to get headers. That’s not
reliable to mix with the `.tbd` Apple ships, and pulls OpenLDAP into
the bootstrap unnecessarily.

We continue to strip the `.dylib`s to avoid accidental dependencies
on the old version. Packages can still link to Apple’s version with
`-framework LDAP`. As very little seems to want the system version,
hopefully this won’t have any major compatibility impact.

emilazy commented:
This should be ready now; it successfully bootstraps and builds Meson (with tests) after the bootstrap. cc @reckenrode

paparodeo reviewed (approved):
LGTM. I don't see `openldap` listed in the requisites of darwin stdenv and doesn't cause a rebuild so seems fine to remove -- there are a bunch of other unneeded attributes in that list as well (brotli, cpio, Cyrus_ssl, ed, expat, groff, kyua, libedit, openssh, patchutils, pbzx, python3Minimal, scons, serf, sqlite, subversion and unzip) can also be removed without triggering a rebuild on aarch64-darwin.

alyssais reviewed (approved):
Happy with the Meson changes. No opinion on Darwin stdenv/SDK changes.

reckenrode reviewed (approved):
Darwin changes LGTM.

toonn commented:
We have achieved a Mexican standoff of approvals : )

@paparodeo, I'm surprised pbzx wouldn't be needed by the stdenv, isn't that what we use to extract the SDK anymore?

emilazy commented:
I guess we might only use the `pbzx` from the bootstrap tools, in which case it wouldn’t need building again? I’ll try to remember to look at the other redundant things another time.

paparodeo commented:
> @paparodeo, I'm surprised pbzx wouldn't be needed by the stdenv, isn't that what we use to extract the SDK anymore?

it is possible I was misinterpreting what stage1Packages was doing. pbzx is used by the bootstrap as it is in stdenv requisites tho I was just removing various packages from stage1Packages and observing which ones didn't trigger a rebuild when removed.

reckenrode commented:
I wouldn’t expect removing `pbzx` to cause a rebuild in most cases. It’s used in the `fetchsdk` FOD. The only time it would is when the source SDK is uncached.
